modules/renter/repair.go

```diff
@@ -253,14 +268,31 @@ func (r *Renter) managedRepairIteration(rs *repairState) {
 	var chunksToDelete []chunkID
 	for chunkID, chunkStatus := range rs.incompleteChunks {
 		// check if the chunk is currently being downloaded for recovery
-		if _, downloading := rs.downloadingChunks[chunkID]; downloading {
-			continue
+		dc, downloading := rs.downloadingChunks[chunkID]
+		if downloading {
+			// Check if download finished
+			select {
+			default:
+				// Check if download timed out
+				if time.Now().After(dc.startTime.Add(chunkDownloadTimeout)) {
+					r.log.Println("Download of chunk to repair timed out")
+					chunksToDelete = append(chunksToDelete, chunkID)
+					delete(rs.downloadingChunks, chunkID)
```

Comment:
Do we need to fail the download here, or can we safely assume that the download was failed when the chunk failed?

Reply:
As far as I can tell it doesn't make a difference. Letting it fail will only set a boolean value and close the channel. Since we also delete it from `rs.downloadingChunks` nothing is holding a reference to that download anyway. I am still going to let it fail for completeness though.